connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -303,4 +307,31 @@ private Endpoint setUpITEndpoint() throws Exception {
 
         return httpComponent.createEndpoint(String.format("https:%s", this.emailConnectorConfig.getItUserServiceURL()));
     }
+
+    /**
+     * Creates the endpoint for the BOP service. It makes Apache Camel trust
+     * BOP service's certificate.
+     * @return the created endpoint.
+     */
+    protected Endpoint setUpBOPEndpoint() {
+        final TrustManagersParameters trustManagersParameters = new TrustManagersParameters();
+        trustManagersParameters.setTrustManager(new BOPTrustManager());
+
+        final SSLContextParameters sslContextParameters = new SSLContextParameters();
+        sslContextParameters.setTrustManagers(trustManagersParameters);
+
+        // Remove the schema from the url to avoid the "ResolveEndpointFailedException".
+        final String fullURL = this.emailConnectorConfig.getBopURL();
+        final String noProtocolURL;
+        if (fullURL.startsWith("https")) {
+            noProtocolURL = fullURL.replace("https://", "");
+        } else {
+            noProtocolURL = fullURL.replace("http://", "");
+        }
+
+        return https(noProtocolURL)
```

Comment:
Is it okay to build the endpoint with `https()` when fullURL starts with `http://`?

Reply:
Good point